### ISSUE TITLE
Boot loop when trying to restore just some of the chats

### ISSUE THREAD

**Issue by danielzyp** [residentsummer/watoi#33]
Hey, this is a bit off topic because I tried to modifie the code, but maybe you guys can help

I tried to modify line 276 by changing 
```
NSArray * androidChats = [self executeQuery:@"SELECT * FROM chat_list"];
```
to
```
NSArray * androidChats = [self executeQuery:@"SELECT * FROM chat_list WHERE key_remote_jid == 'NUMBER@s.whatsapp.net' OR "key_remote_jid == 'NUMBER@s.whatsapp.net' OR (subject IS NOT NULL AND length(subject) > 0)];
```
where `NUMBER` is the jid of my contacts, for about 100 contacts
I tested the query before on the Android db file and it worked just as expected

It seemed like working just fine until I actually went to restore the backup and then got stuck in a boot loop
I had the exact same amount of "Imported chats XXX" logs as the amount of contacts+groups
When I tried to restore without the code modification it worked fine, not great - but fine (it didnt order the contacts by last activity)
Using latest Whatsapp with the corresponding IPA

Any suggestions on how to achieve what I want?
Basically I want to restore groups and contacts only, and I got a list of contacts.

**Comment by residentsummer** [comment]
Hi, @danielzyp ! Did you tried to restore everything? It worked?

**Comment by danielzyp** [comment]
> Hi, @danielzyp ! Did you tried to restore everything? It worked?

Yes, restoring everything worked, it wasn't great but it worked.
I eventually ended up using a shelf product called `Backuptrans Android WhatsApp to iPhone Transfer` so I won't be pursuing this feature.
I would like to thank you anyways for this great source, I learned a lot about how Whatsapp database works from your code.

**Comment by residentsummer** [comment]
Ok, @danielzyp, thank you for the feedback and kind words. That's very strange that _contents_ of a file may lead to a bootloop. Looks like there is still some flaky issue in the restore (or backup editing) process itself.
